api/res/openapi.yml

```diff
@@ -131,6 +125,42 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/ProblemDetails'
+  /apps/{appName}/logs/{serviceName}/:
+    get:
+      summary: Retrieves the logs from stdout/stderr of the specified container.
+      parameters:
+        - $ref: '#/components/parameters/appName'
+        - in: path
+          name: serviceName
+          description: Name of the service
+          required: true
+          schema:
+            type: string
+        - in: query
+          name: since
+          description: Date time since when the logs have to retrieved. By default the logs from the beginning are crawled.
+          schema:
+            type: string
+            format: date-time
+          example: '2019-07-22T08:42:47-00:00'
+        - in: query
+          name: limit
```

Comment:
I was surprised that the limit is a limit on the first `n` lines rather than the last `n` lines. Is it possible to the most current `n` lines also? IMHO this will be a common use case.

Reply:
I assume the purpose of the limit parameter is to reduce the memory footprint. But if you want to apply it to the last n lines instead of the first n lines, the whole content of the log file has to be loaded into memory first, before the respective starting line can be determined.

Reply:
You only need a ring-buffer of `n` lines to calculate the last `n` lines. When I wrote the comment, I overlooked the `since`-parameter, which already covers the mentioned use-case, so I don't think a limit for the last `n`-lines is needed anymore.

Reply:
The purpose of the `limit` parameter is to make the REST calls for the frontend simple. The frontend request the first chunk of logs and the REST API responsed with the first chunk and a link header (see [RFC5988](https://tools.ietf.org/html/rfc5988)) to the next page (a.k.a. the next chunk of logs). When the user scrolled to the end of logs, the frontend can use the next link to crawl more logs.